core/src/com/unciv/ui/worldscreen/WorldScreen.kt

```diff
@@ -572,19 +572,29 @@ class WorldScreen(val viewingCiv:CivilizationInfo) : CameraStageBaseScreen() {
             return
 
         // remove current listener for the "BACK" button to avoid showing the dialog twice
-        stage.removeListener( backButtonListener )
+        stage.removeListener (backButtonListener)
+
+        // Since Popups including the Main Menu and the Options screen have no own back button
+        // listener and no trivial way to set one, back/esc with one of them open ends up here.
+        // Also, the reaction of other popups like 'disband this unit' to back/esc feels nicer this way.
+        // After removeListener just in case this is slow (enumerating all stage actors)
+        if (hasOpenPopups()) {
+            closeAllPopups()
+            stage.addListener (backButtonListener)
```

Comment:
Why do you need to remove the listener and add it back again?
Doesn't make sense to check for open popups _before_ removing the listener?

Reply:
I commented specifically on that... Just in case hasOpenPopups() is expensive. There's an iterator over Gdx-internal stuff in there. I mean, if you're right - why did it have a spam-clicker guard in the first place?

Reply:
Oh, that wasn't a guard against repeated events arriving too fast - OK, then let's have the Back/Esc keys open _and_ close the 'Exit?' prompt.